scipy/spatial/_spherical_voronoi.py

```diff
@@ -141,33 +129,31 @@ class SphericalVoronoi:
 
     """
 
-    def __init__(self, points, radius=1, center=(0, 0, 0), threshold=1e-06):
+    def __init__(self, points, radius=1, center=None, threshold=1e-06):
+
+        self.points = np.array(points).astype(np.double)
```

Comment:
How is this change related to the expanding dimensionality? The argument is already defined as an array of doubles, right? The coercion to support array-like is a separate matter, no?

Reply:
In 3D `center=(0,0,0)` makes sense.
In 4D we need `center=(0,0,0,0)`
In 6D we need `center=(0,0,0,0,0,0)` etc.

We can't have variable length default parameters, so I had to revert to `None`.

Reply:
I mean the new line with `self.points = np.array(points).astype(np.double)`.

Reply:
Not saying it is a bad idea to support array-like input eventually, just saying I don't immediately see the connection with higher dimensional support.

Reply:
> I mean the new line with self.points = np.array(points).astype(np.double).

Ah sorry, I misunderstood.  You are right - there is no specific connection to higher-dimensional input.  If the input is an integer array though, then `self.points` inherits this type, and it is inconsistent with the attributes in the documentation.  This is perhaps a minor detail though.

If/when #10759 is merged, we will want to copy `points` rather than just having a reference, since otherwise the user could change the values of `points` before calling `calculate_areas` and get an inconsistent result.

Reply:
We should probably remove it then, and if there's pathological input that causes a bug perhaps deal with that in a separate PR along with a test.

Reply:
OK. Fixed!